From 058765805d4a8cbe7ec7f6c31c14e8953982e4b5 Mon Sep 17 00:00:00 2001
From: Patrick Delaunay <patrick.delaunay@foss.st.com>
Date: Fri, 29 Nov 2024 18:28:09 +0100
Subject: [PATCH 1518/1695] plat-stm32mp1: clean definition of teeram_bkp_mm

Move the definition of the static variable teeram_bkp_mm at beginning
of the file.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git f207c9d4b464c7466f0af97b1432c56c37976604 plat-stm32mp1: clean definition of teeram_bkp_mm]
Fixes: cf483e4784e5 ("plat-stm32mp1: add low power drivers for stm32mp1")
Signed-off-by: Patrick Delaunay <patrick.delaunay@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/423280
Change-Id: I4e442c67563d72a109ee70129691b6417b14e078
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/437076
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
---
 core/arch/arm/plat-stm32mp1/pm/context.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/core/arch/arm/plat-stm32mp1/pm/context.c b/core/arch/arm/plat-stm32mp1/pm/context.c
index e48e8e469..f323e9bdc 100644
--- a/core/arch/arm/plat-stm32mp1/pm/context.c
+++ b/core/arch/arm/plat-stm32mp1/pm/context.c
@@ -132,6 +132,8 @@ struct pm_mailbox {
 #endif
 };
 
+static tee_mm_entry_t *teeram_bkp_mm;
+
 /*
  * BKPSRAM contains OP-TEE resume instruction sequence which restores
  * TEE RAM content. The BKPSRAM contains restoration materials
@@ -231,8 +233,6 @@ void stm32mp_pm_wipe_context(void)
 	clk_disable(pm_clocks.bkpsram);
 }
 
-static tee_mm_entry_t *teeram_bkp_mm;
-
 static void init_retram_resume_resources(void)
 {
 	struct retram_resume_ctx *ctx = get_retram_resume_ctx();
-- 
2.47.2

